conans/client/tools/settings.py

```diff
@@ -49,6 +49,9 @@ def deduced_cppstd():
         return cppstd_default(compiler, compiler_version)
 
     current_cppstd = deduced_cppstd()
+    if current_cppstd is None:
+        raise ConanException("Could not detect the current default cppstd because "
```

Comment:
I'd appreciate if we could have a different exception type to be able to catch this exact situation

Reply:
Yeah, that makes sense, I gonna update it.

Reply:
Thank you!

Reply:
I think raising a ``ConanInvalidConfiguration`` is the way to go. Please @uilianries update this, I am assigning it to 1.24.

Reply:
@memsharded Done!